src/utils/ucc_lock_free_queue.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_LOCKFREE_QUEUE_H_
+#define UCC_LOCKFREE_QUEUE_H_
+
+#include "utils/ucc_spinlock.h"
+#include "utils/ucc_atomic.h"
+#include <string.h>
+
+/* This data structure is thread safe */
+
+// Number of elements in a single lock free pool - could be changed, but in tests performed great
+#define LINE_SIZE 8
+#define NUM_POOLS 2
+
+typedef struct ucc_lf_queue_elem {
+    uint8_t         was_progressed;
```

Comment:
name "was_progresses" does not fit. It is generic lock-free queue implementation and it has no notion of "progress" - plz rename. Maybe "was_requeued" or smth?

Reply:
Fixed